sub-process.c

```diff
@@ -126,6 +127,53 @@ int subprocess_start(struct hashmap *hashmap, struct subprocess_entry *entry, co
 	return 0;
 }
 
+int subprocess_start_argv(struct hashmap *hashmap,
```

Comment:
It would be possible to pass all of the command line args over stdin/out, but a pain.

The main reason (and the reason I needed it when doing gvfs-helper) is that the
helper is a builtin command, and I need platform help to find the correct git.exe
executable associated with the currently running instance of Git, so I need the argv
form to allow run-process to take care of the details as it does for other (short-lived)
builtin child commands.

I'll reword in my next version.